### PR TITLE
mgr/balancer: auto balance a list of pools

### PR DESCRIPTION
**Pull request by xiexingguo**
Add support which limits the auto balancer to a comma-separated list of pools.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -388,6 +412,59 @@ def handle_command(self, inbuf, command):
...
+                if p in pool_name_by_id:
+                    final_ids.append(p)
+                    final_names.append(pool_name_by_id[p])
+                else:
+                    should_prune = True
+            if should_prune: # some pools were gone, prune
+                if final_ids:
+                    self.set_module_option('pool_ids', ','.join(final_ids))
+                else:
+                    self.set_module_option('pool_ids', '')
```
Comment (sebastian-philipp) [inline src/pybind/mgr/balancer/module.py]: `','.join(final_ids)` is already the empty string: 

```python
In [1]: ','.join([])
Out[1]: ''
```

File: src/pybind/mgr/balancer/module.py [hunk 2]
```diff
@@ -388,6 +412,59 @@ def handle_command(self, inbuf, command):
...
+            osdmap = self.get_osdmap()
+            pool_ids = [str(p['pool']) for p in osdmap.dump().get('pools', [])]
+            pool_id_by_name = dict((p['pool_name'], p['pool']) for p in osdmap.dump().get('pools', []))
+            final = [p for p in existing if p in pool_ids]
+            to_delete = [str(pool_id_by_name[p]) for p in raw_names if p in pool_id_by_name]
+            final = set(final) - set(to_delete)
+            if final:
+                self.set_module_option('pool_ids', ','.join(final))
+            else:
+                self.set_module_option('pool_ids', '')
```
Comment (sebastian-philipp) [inline src/pybind/mgr/balancer/module.py]: see above

File: src/pybind/mgr/balancer/module.py [hunk 3]
```diff
@@ -483,7 +560,22 @@ def serve(self):
...
+                final = []
+                if allow is not '':
+                    allow = allow.split(',')
+                    valid = [str(p['pool']) for p in osdmap.dump().get('pools', [])]
+                    final = set(allow) & set(valid)
+                    if set(allow) - set(valid): # some pools were gone, prune
+                        if final:
+                            self.set_module_option('pool_ids', ','.join(final))
+                        else:
+                            self.set_module_option('pool_ids', '')
```
Comment (sebastian-philipp) [inline src/pybind/mgr/balancer/module.py]: see above

File: src/pybind/mgr/balancer/module.py (hunk 1 above)
Comment (xiexingguo) [reply src/pybind/mgr/balancer/module.py]: @sebastian-philipp Thanks!

**Comment by liewegas** [comment]
retest this please

**Comment by xiexingguo** [comment]
retest this please

**Pull request merged**